File: Apps/BankBase/backend/src/main/java/org/d3javu/backend/kafka/requests/client/ClientCreateRequest.java
package org.d3javu.backend.kafka.requests.client;

import org.d3javu.backend.dto.client.ClientCreateRecord;

import java.time.LocalDate;

public record ClientCreateRequest(String surname, String name, String patronymic, LocalDate dateOfBirth,
                                  String phoneNumber, String email, String passwordHash) {

    public ClientCreateRequest(ClientCreateRecord clientCreateRecord, String passwordHash) {
        this(
                clientCreateRecord.surname(),
                clientCreateRecord.name(),
                clientCreateRecord.patronymic(),
                clientCreateRecord.dateOfBirth(),
                clientCreateRecord.phoneNumber(),
                clientCreateRecord.email(),
                passwordHash
        );
    }
}

File: Apps/BankBase/backend/src/main/scala/internalkafka/messages/client/ClientCreateRequest.scala
package internalkafka.messages.client

import zio.json.{DeriveJsonCodec, DeriveJsonDecoder, DeriveJsonEncoder, JsonCodec, JsonDecoder, JsonEncoder}

import java.time.LocalDate

case class ClientCreateRequest(
                                surname: String,
                                name: String,
                                patronymic: String,
                                dateOfBirth: LocalDate,
                                phoneNumber: String,
                                email: String,
                                passwordHash: String,
                              )

object ClientCreateRequest {
  inline given JsonCodec[ClientCreateRequest] = DeriveJsonCodec.gen[ClientCreateRequest]
}
